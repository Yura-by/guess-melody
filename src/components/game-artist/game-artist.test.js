import React from 'react';
import renderer from 'react-test-renderer';
import GameArtist from './game-artist';

const mock = {
  type: `artist`,
  song: {
    artist: `Mikle Jackson`,
    src: `https://upload.wikimedia.org/wikipedia/commons/6/64/Ugandan_national_anthem%2C_performed_by_the_U.S._Navy_Band.ogg`,
  },
  answers: [
    {
      picture: `https://via.placeholder.com/134x134`,
      artist: `Mikle Jackson`,
    },
    {
      picture: `https://via.placeholder.com/134x134`,
      artist: `Selena Gomez`,
    },
    {
      picture: `https://via.placeholder.com/134x134`,
      artist: `50 cent`,
    },
  ],
};

it(`GameArtist snaphot test`, () => {
  const gameArtistTree = renderer
    .create(<GameArtist
      question={mock}
      onAnswer={() => {}}
      questionNumber={0}
    />)
    .toJSON();

  expect(gameArtistTree).toMatchSnapshot();
});
