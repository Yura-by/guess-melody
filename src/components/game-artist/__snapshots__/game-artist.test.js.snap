// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameArtist snaphot test 1`] = `
<section
  className="game game--artist"
>
  <header
    className="game__header"
  >
    <a
      className="game__back"
      href="#"
    >
      <span
        className="visually-hidden"
      >
        Сыграть ещё раз
      </span>
      <img
        alt="Угадай мелодию"
        className="game__logo"
        src="img/melody-logo-ginger.png"
      />
    </a>
    <svg
      className="timer"
      viewBox="0 0 780 780"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        className="timer__line"
        cx="390"
        cy="390"
        r="370"
      />
    </svg>
    <div
      className="timer__value"
      xmlns="http://www.w3.org/1999/xhtml"
    >
      <span
        className="timer__mins"
      >
        05
      </span>
      <span
        className="timer__dots"
      >
        :
      </span>
      <span
        className="timer__secs"
      >
        00
      </span>
    </div>
    <div
      className="game__mistakes"
    >
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
    </div>
  </header>
  <section
    className="game__screen"
  >
    <h2
      className="game__title"
    >
      Кто исполняет эту песню?
    </h2>
    <div
      className="game__track"
    >
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          type="button"
        />
        <div
          className="track__status"
        >
          <audio />
        </div>
      </div>
    </div>
    <form
      className="game__artist"
      onChange={[Function]}
    >
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="artist-0"
          name="answer"
          type="radio"
          value="artist-0"
        />
        <label
          className="artist__name"
          htmlFor="artist-0"
        >
          <img
            alt="Mikle Jackson"
            className="artist__picture"
            src="https://via.placeholder.com/134x134"
          />
          Mikle Jackson
        </label>
      </div>
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="artist-1"
          name="answer"
          type="radio"
          value="artist-1"
        />
        <label
          className="artist__name"
          htmlFor="artist-1"
        >
          <img
            alt="Selena Gomez"
            className="artist__picture"
            src="https://via.placeholder.com/134x134"
          />
          Selena Gomez
        </label>
      </div>
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="artist-2"
          name="answer"
          type="radio"
          value="artist-2"
        />
        <label
          className="artist__name"
          htmlFor="artist-2"
        >
          <img
            alt="50 cent"
            className="artist__picture"
            src="https://via.placeholder.com/134x134"
          />
          50 cent
        </label>
      </div>
    </form>
  </section>
</section>
`;
